language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Exception.hs

```diff
@@ -54,10 +153,23 @@ instance ( PrettyBy config (Term TyName Name ())
     prettyBy _      (OtherMachineError err)               =
         pretty err
 
-instance Pretty err => Show (MachineException err) where
-    show (MachineException err cause) = fold
-        [ "An abstract machine failed: ", docString $ prettyPlcReadableDebug err, "\n"
-        , "Caused by: ", docString $ prettyPlcReadableDebug cause
+instance (Pretty internal, Pretty user, PrettyBy config (Term TyName Name ())) =>
+            PrettyBy config (EvaluationError internal user) where
+    prettyBy config (InternalEvaluationError err) = fold
+        [ "An internal error has occurred:", "\n"
+        , prettyBy config err
+        ]
+    prettyBy _      (UserEvaluationError err) = fold
+        [ "A user error has occurred:", "\n"
```

Comment:
Use `line` instead of `"\n"` (or `hardline` if you feel like it) - helps the algorithm make proper decisions.